cmd/uniq/uniq.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+)
+
+type options struct {
+	printDuplicates bool
+	printEmptyLines bool
+	printLineNumber bool
+}
+
+func parseArgs(args []string) options {
+	var opts options
+
+	for _, opt := range args[1:] {
+		switch opt {
+		case "-dup":
+			opts.printDuplicates = true
+		case "-empty":
+			opts.printEmptyLines = true
+		case "-num":
+			opts.printLineNumber = true
+		default:
+			fmt.Println("Usage:")
+			fmt.Println("uniq [-dup | -empty | -num]")
+			os.Exit(1)
+		}
+	}
+	return opts
+}
+
+func uniq(input *os.File, opts options) {
+	inputMap := make(map[string]*string)
+	lineNumsMap := make(map[*string][]int)
+	lineMap := make(map[int]*string)
+
+	reader := bufio.NewReader(input)
+	i := 0
+	for {
+		i++
+		lineBytes, err := reader.ReadBytes(byte('\n'))
+		if err == io.EOF {
+			break
+		}
+
+		line := string(lineBytes)
+		var lineP *string
+		if val, has := inputMap[line]; has {
+			lineP = val
+		} else {
+			lineP = &line
+		}
+		lineCount := len(lineNumsMap[lineP])
```

Comment:
I see. But in this case. lineCount is the count of lines not the length of a line. It can be confusing to use lineLen.